### PR TITLE
Refactor error and "no preview" views into core

### PR DESCRIPTION
**Pull request by Hypnosphi**
After this change, `preview` part in each app contains only framework-specific things

### CODE REVIEW

File: lib/core/src/client/preview/start.js [hunk 1]
```diff
@@ -11,6 +12,50 @@ import reducer from './reducer';
...
+
+function showNopreview() {
+  document.body.classList.remove(classes.MAIN);
+  document.body.classList.remove(classes.ERROR);
+
+  document.body.classList.add(classes.NOPREVIEW);
+}
+
+function showErrorDisplay({ message, stack }) {
+  document.getElementById('error-message').textContent = message;
```
Comment (ndelangen) [inline lib/core/src/client/preview/start.js]: Interesting, we're just using plain JS here to show/hide plain messages.

Makes sense.. I always assumed the error handling and debugging experiences for the different frameworks would be wildly different and thus would warrant having a separate component for each.

But setup is quite nice and simple I guess, description has to be an html-string I guess, and so a framework could submit some rich data down to the error display.

**Review by ndelangen — APPROVED** [review]

**Comment by ndelangen** [comment]
Would like approval of some more people

File: lib/core/src/client/preview/start.js (hunk 1 above)
Comment (Hypnosphi) [reply lib/core/src/client/preview/start.js]: > description has to be an html-string

Well right now it isn't supported (I use `textContent` not `innerHTML`). Error stack can be user-generated, so there's some room for xss here

Comment (Hypnosphi) [reply lib/core/src/client/preview/start.js]: But they were enough so far

File: app/angular/package.json [hunk 2]
```diff
@@ -38,26 +35,19 @@
     "babel-runtime": "^6.23.0",
     "case-sensitive-paths-webpack-plugin": "^2.1.2",
     "core-js": "^2.5.5",
-    "cross-env": "^5.1.4",
```
Comment (igor-dv) [inline app/angular/package.json]: So this : `"dev": "cross-env DEV_BUILD=1 nodemon -e js,ts --watch ./src --exec \"yarn prepare\"",` is not needed anymore (I think it wasn't in use before anyway) ?

**Review by igor-dv — APPROVED** [review]
Very cool!

File: app/angular/package.json (hunk 2 above)
Comment (Hypnosphi) [reply app/angular/package.json]: It's executed in root, so no need to have those dependencies on this level

Comment (Hypnosphi) [reply app/angular/package.json]: Or maybe it's not, then we can remove the script as well

Comment (igor-dv) [reply app/angular/package.json]: I think it's not (maybe it was, but a long time ago). 
The last time I've been touching there was due to the ts distribution issue, and I've cleaned thing there. 
But maybe I am missing something

Comment (igor-dv) [reply app/angular/package.json]: Oh, I am late

File: lib/core/src/client/preview/start.js (hunk 1 above)
Comment (tmeasday) [reply lib/core/src/client/preview/start.js]: What is XSS in this context?

Comment (tmeasday) [reply lib/core/src/client/preview/start.js]: As in who is attacking whom?

**Comment by tmeasday** [comment]
This is great! Thanks for this @Hypnosphi

File: lib/core/src/client/preview/start.js (hunk 1 above)
Comment (Hypnosphi) [reply lib/core/src/client/preview/start.js]: Somebody giving you some tricky URL when you have public storybook on some route of your main website whicj has auth and cookies

Comment (tmeasday) [reply lib/core/src/client/preview/start.js]: Oooh, that's a good one, touche

**Pull request merged**